web/src/stream_data.ts

```diff
@@ -177,6 +179,20 @@ export function get_sub(stream_name: string): StreamSubscription | undefined {
     return undefined;
 }
 
+export function get_sub_by_id_string(stream_id_string: string): StreamSubscription {
+    const stream = get_possible_sub_by_id_string(stream_id_string);
+    assert(stream !== undefined);
+    return stream;
+}
+
+export function get_possible_sub_by_id_string(
+    stream_id_string: string,
+): StreamSubscription | undefined {
+    const stream_id = Number.parseInt(stream_id_string, 10);
+    const stream = stream_info.get(stream_id);
+    return stream;
+}
```

Comment:
These should be in the other order: Usually best to call a function you've already defined. 

Also, I think the naming scheme of having the one that can fail with `assert_valid` in its name, rather than `possible` in the name of the one that can return undefined, would be a lot more consistent with other function names.

Reply:
What about `get_valid_sub_by_id_string`? It's a bit shorter and more readable in my opinion.

Reply:
I don't like it as much (it doesn't speak as clearly that the caller is claiming the ID string is valid, and thus we expect to be guaranteed to get a sub back; it sounds like we're asking for a "valid sub" instead) but we can merge this and then bikeshed in #frontend.

Reply:
https://chat.zulip.org/#narrow/stream/6-frontend/topic/naming.20functions.20that.20don't.20return.20undefined/near/1944622